### PR TITLE
Add integration tests to empty plugin.

### PR DESCRIPTION
**Pull request by christ66**
To test this run:

```
mvn clean install archetype:integration-test
```

### CODE REVIEW

**Comment by christ66** [comment]
@jglick Can you run this locally? I can run the integration tests locally without any issues, but Jenkins shows failures creating the empty directories.

File: empty-plugin/src/test/resources/projects/testInstall/archetype.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+groupId=com.test
```
Comment (jglick) [inline empty-plugin/src/test/resources/projects/testInstall/archetype.properties]: should be able to omit

File: empty-plugin/src/test/resources/projects/testInstall/archetype.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+artifactId=testArtifact
+version=1.0-SNAPSHOT
+package=test
```
Comment (jglick) [inline empty-plugin/src/test/resources/projects/testInstall/archetype.properties]: ditto

File: empty-plugin/src/test/resources/projects/testInstall/archetype.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+groupId=com.test
+artifactId=testArtifact
+version=1.0-SNAPSHOT
```
Comment (jglick) [inline empty-plugin/src/test/resources/projects/testInstall/archetype.properties]: ditto

File: empty-plugin/src/test/resources/projects/testInstall/reference/pom.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jglick) [inline empty-plugin/src/test/resources/projects/testInstall/reference/pom.xml]: Would actually rather not have the reference file—just forces everyone to do every change to the archetype twice. Suffices to verify that it builds.

File: empty-plugin/pom.xml [hunk 5]
```diff
@@ -9,4 +9,18 @@
...
     <name>Empty Jenkins Plugin</name>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <version>2.3</version>
+                <configuration>
+                    <!-- Required for testing empty directories -->
```
Comment (jglick) [inline empty-plugin/pom.xml]: What is the purpose of this? The archetype was working fine without it.

File: Jenkinsfile [hunk 6]
```diff
@@ -12,17 +12,15 @@ pipeline {
     stages {
         stage('main') {
             steps {
-                sh 'mvn -B clean verify'
+                sh 'mvn -B clean install archetype:integration-test'
```
Comment (jglick) [inline Jenkinsfile]: Or could go back to `verify` if there were a plugin execution for `archetype:integration-test` in `pom.xml`.

File: Jenkinsfile [hunk 7]
```diff
@@ -12,17 +12,15 @@ pipeline {
...
             }
-            /* TBD archetype:integration-test:
             post {
                 failure {
                     catchError { // JENKINS-42478: in case the failure occurred prior to getting the node
-                        archiveArtifacts artifacts: 'target/its/ * /build.log', allowEmptyArchive: true
+                        archiveArtifacts artifacts: '**/target/test-classes/**/*', allowEmptyArchive: true
```
Comment (jglick) [inline Jenkinsfile]: AFAICT none of these files are useful for diagnosis at all.

**Comment by jglick** [comment]
See #4.
